examples/scripts/ct_astra_modl_train_foam2.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+r"""
+CT Reconstruction with MoDL
+===========================
+
+This example demonstrates the training and application of a model-based deep learning
+(MoDL) architecture described in :cite:`aggarwal-2019-modl`
+ applied to a CT reconstruction problem.
+
+The source images are foam phantoms generated with xdesign.
+
+A class [flax.MoDLNet](../_autosummary/scico.learning.rst#scico.learning.MoDL)
+ implements the MoDL architecture, which solves the optimization problem
+
+  $$\mathrm{argmin}_{\mathbf{x}} \; \| A \mathbf{x} - \mathbf{y} \|_2^2 + \lambda \, \| \mathbf{x} - \mathrm{D}_w(\mathbf{x})\|_2^2 \;,$$
+
+where $A$ is a tomographic projector, $\mathbf{y}$ is a set of sinograms, $\mathrm{D}_w$ is the
+ regularization (a denoiser), and $\mathbf{x}$ is the set of reconstructed images.
+  The MoDL abstracts the iterative solution by an unrolled network where each iteration corresponds
+  to a different stage in the MoDL network and updates the prediction by solving
+
+  $$\mathbf{x}^{k+1} = (A^T A + \lambda \, \mathbf{I})^{-1} (A^T \mathbf{y} + \lambda \, \mathbf{z}^k) \;,$$
+
+via conjugate gradient. In the expression, $k$ is the index of the stage (iteration),
+ $\mathbf{z}^k = \mathrm{ResNet}(\mathbf{x}^{k})$ is the regularization
+ (a denoiser implemented as a residual convolutional neural network), $\mathbf{x}^k$ is the output
+  of the previous stage, $\lambda > 0$
+  is a learned regularization parameter, and $\mathbf{I}$ is the identity operator.
+  The output of the final stage is the set of reconstructed images.
+"""
+
+import os
+from functools import partial
+from time import time
+
+import numpy as np
+
+import jax
+
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+from scico import flax as sflax
+from scico import metric, plot
+from scico.flax.examples import load_ct_data
+from scico.flax.train.train import clip_positive, construct_traversal
+from scico.linop.radon_astra import TomographicProjector
+
+"""
+Prepare parallel processing. Set an arbitrary processor
+count (only applies if GPU is not available).
+"""
+os.environ["XLA_FLAGS"] = "--xla_force_host_platform_device_count=8"
+platform = jax.lib.xla_bridge.get_backend().platform
+print("Platform: ", platform)
+
+
+"""
+Read data from cache or generate if not available.
+"""
+N = 256  # phantom size
+train_nimg = 536  # number of training images
+test_nimg = 64  # number of testing images
+nimg = train_nimg + test_nimg
+n_projection = 45  # CT views
+
+trdt, ttdt = load_ct_data(train_nimg, test_nimg, N, n_projection, verbose=True)
+
+
+"""
+Build CT projection operator.
+"""
+angles = np.linspace(0, np.pi, n_projection)  # evenly spaced projection angles
+A = (
+    1
+    / N
```

Comment:
Clearer as `(1.0/N) * ...`?

Reply:
What a weird expression. Changed to suggested form.

Reply:
How about
```python
A = TomographicProjector(...)
A = (1/N) * A
```

Reply:
I implemented this suggestion.